### ISSUE TITLE
lein repl out of memory

### ISSUE THREAD

**Issue by seagreen** [technomancy/leiningen#2288]
## Issue description

`lein repl` fails with a "cannot allocate memory" error, even though plenty of memory (about 5 gigs) is free.

### Steps to reproduce

```zsh
[~/code/mine/clojure]traveller@nivian $ nix-shell -p leiningen

[nix-shell:~/code/mine/clojure]$ lein repl
#
# There is insufficient memory for the Java Runtime Environment to continue.
# Native memory allocation (malloc) failed to allocate 32744 bytes for ChunkPool::allocate
# An error report file with more information is saved as:
# /home/traveller/code/mine/clojure/hs_err_pid28042.log
#
# Can't open file to dump replay data. Error: Cannot allocate memory
```

This creates a file with more error info, which [I've posted here](https://gist.github.com/seagreen/47c41c080ea4037ea0faae10dcccfc3b).

## Technical details

* `lein --version`: Leiningen 2.7.1 on Java 1.8.0_131 OpenJDK 64-Bit Server VM
* System: NixOS 17.09.git.0afb6d7 (Hummingbird)

**Comment by technomancy** [comment]
To be clear, are you getting this error when you run `lein repl` outside a project, or only when it's inside a project directory?

**Comment by seagreen** [comment]
Good question, I'm getting it in both cases.

**Comment by seagreen** [comment]
This turns out to be a more interesting situation than I realized.

It was on my laptop where I first ran into the issue with `lein repl`. I SSH'd to my desktop, verified it was having the problem too, and then wrote up the bug report above copying commands and output from my desktop.

_But_ by mistake I got the output of `lein --version` for the bug report from my laptop, not my desktop. Now I realize that every `lein` command is failing for my desktop with that same error, not just `lein repl`, but also `lein new`, `lein version`, etc.

(Since I can't run `lein version` on my desktop, I looked into my package manager to verify that the version is still 2.7.1, and it is.)

Meanwhile, to add to the weirdness, `lein repl` has spontaneously started working on my laptop.

I have no idea how to interpret this, but I didn't want to send you down the wrong path looking into REPL code if the problem's not there. If there's any other info I can provide let me know.
